packages/inter-protocol/test/vaultFactory/test-math.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable no-lone-blocks */
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { test } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+
+import { makeIssuerKit } from '@agoric/ertp';
+import { maxDebtForVault } from '../../src/vaultFactory/math.js';
+import { withAmountUtils } from '../supports.js';
+
+const stable = withAmountUtils(makeIssuerKit('Stable'));
+const aeth = withAmountUtils(makeIssuerKit('Aeth'));
+
+/**
+ * Max debt for a fixed collateral of 1_000 aeth
+ *
+ * @param {*} t
+ * @param {readonly [Number, Number, Number]} input
+ * @param {bigint} result
+ */
+function checkMax(
+  t,
+  [pricePerColl, liquidationMarginNum, minimumCollateralizationNum],
+  result,
+) {
+  const costOfCollateral = BigInt(1_000 * pricePerColl);
+
+  /** @type {PriceQuote} */
+  const quote = {
+    // @ts-expect-error cast
+    quoteAmount: { value: [{ amountOut: stable.make(costOfCollateral) }] },
+  };
+
+  // the liquidationMargin term is a ratio so we add 1 to our incoming parameter
+  const liquidationMargin = harden({
+    numerator: stable.make(1_000n + BigInt(liquidationMarginNum * 1_000)),
+    denominator: aeth.make(1_000n),
+  });
+
+  const minimumCollateralization = harden({
+    numerator: stable.make(
+      1_000n + BigInt(minimumCollateralizationNum * 1_000),
+    ),
+    denominator: aeth.make(1_000n),
+  });
+
+  t.is(
+    maxDebtForVault(quote, liquidationMargin, minimumCollateralization).value,
+    result,
+  );
+}
+
+test('zero liquidation margin', checkMax, [1.0, 0, 0], 1000n);
+test('5% liquidation margin', checkMax, [1.0, 0.05, 0], 952n);
+test('100% liquidation margin', checkMax, [1.0, 1.0, 0], 500n);
+
+{
+  // negatives
+  test(
```

Comment:
What are the consequences of putting some test inside a subcontext?

Reply:
No runtime consequences since there are no bindings in the block scope.

The motivation is to group the `negatives` tests together. Other test runners let test be organized hierarchically. AVA won't support this, though they endorse 3rd party libs: https://www.npmjs.com/package/ava-describe. But by a quick look that seemed too complex

Reply:
Is it still possible to invoke them separately from the command line?  (Purely academic curiosity; I use test.only() and run from inside my IDE.)

Reply:
yeah the `{}` just makes a scope but doesn't affect what gets run.  I don't know how selection would work with ava-describe